src/t_stream.c

```diff
@@ -1523,50 +1513,25 @@ uint64_t streamEstimateDistanceFromFirstEverEntry(stream *s, streamID *id) {
         return s->entries_added;
     } else if (cmp_last > 0) {
         /* The counter of a future ID is unknown. */
-        return 0;
+        return ULLONG_MAX;
     }
 
-    int cmp_first = streamCompareID(id,&s->first_id);
-    int cmp_xdel = streamCompareID(id,&s->max_deleted_entry_id);
-    if (streamIDEqZero(&s->max_deleted_entry_id) || cmp_xdel > 0) {
-        /* There's definitely no fragmentation. */
-        if (cmp_first < 0) {
+    int cmp_id_first = streamCompareID(id,&s->first_id);
+    int cmp_xdel_first = streamCompareID(&s->max_deleted_entry_id,&s->first_id);
+    if (streamIDEqZero(&s->max_deleted_entry_id) || cmp_xdel_first < 0) {
+        /* There's definitely no fragmentation ahead. */
+        if (cmp_id_first < 0) {
             /* Return the estimated counter. */
             return s->entries_added - s->length;
-        } else if (cmp_first == 0) {
+        } else if (cmp_id_first == 0) {
             /* Return the exact counter of the first entry in the stream. */
             return s->entries_added - s->length + 1;
-        } else {
-            /* The add counter of an arbitrary ID in the stream is unknown. */
-            return 0;
         }
     }
 
-    /* There's a chance that the next ID is valid, and the only thing
-     * before our ID is a tombstone. If we got here, we know that the
-     * stream isn't empty and that our ID is before its start, so there 
-     * must be at least one entry with a larger ID in it. */
-    streamIterator si;
-    int64_t numfields;
-    streamID next_id = *id;
-    streamIncrID(&next_id);
-    streamIteratorStart(&si,s,&next_id,NULL,0);
-    serverAssert(streamIteratorGetID(&si,&next_id,&numfields));
-    streamIteratorStop(&si);
-
-    int cmp_next = streamCompareID(&next_id,&s->max_deleted_entry_id);
-    if (cmp_next > 0) {
-        /* Return the estimated first counter. At this point we know that
-         * the ID is smaller than the first ID, and that there's a tombstone
-         * between them. However, the tombstone is also before the first ID,
-         * so the next ID is actually the first in the stream. This lets us
-         * set the counter to that of the next ID, but off by one. */
-        return s->entries_added - s->length - 1;
-    }
```

Comment:
why was this removed?

Reply:
I realized it was redundant given s->first_id.